src/esp/physics/RigidBase.h

```diff
@@ -334,11 +334,27 @@ class RigidBase : public esp::physics::PhysicsObjectBase {
   virtual void setRestitutionCoefficient(
       CORRADE_UNUSED const double restitutionCoefficient) {}
 
-  /** @brief Get the scale of the object set during initialization.
+  /** @brief Get the scale of the object.
    * @return The scaling for the object relative to its initially loaded meshes.
    */
   virtual Magnum::Vector3 getScale() const {
-    return initializationAttributes_->getScale();
+    if (scalingNode_ != nullptr) {
+      return scalingNode_->scaling();
+    } else {
+      return initializationAttributes_->getScale();
+    }
+  }
+
+  /** @brief Set the non-uniform scaling of the object.
+   * @param scale The desired non-uniform scaling of the object.
+   */
+  virtual void setScale(const Magnum::Vector3& scale) {
+    if (objectMotionType_ != MotionType::STATIC) {
```

Comment:
Seems confusing to silently fail for STATIC objects and objects without `scalingNode_`.